### PR TITLE
Update bootstrap to beta3

### PR DESCRIPTION
**Pull request by atalargo**
Update bootstrap and boostrap-vue dependency to 4.0.0-beta3 and 1.1.0
- Fix import call (need bottstrap's functions now in scss import)
- Fix some variables
- Fix Dropdown directive
- Fix Progress Bar/Circle Progress Bar scss
- Remove Bulma-vue-expand and bulma and replace it by B4's Collapse

+ transform relative node_modules @import in scss part of vue files of vuestic component to ~[node_modules] form (cf commit 
dc25a63 )

+ Proposal for not relative JS imports for vuestic-components ( cf commit 25ab85a )

### CODE REVIEW

**Comment by atalargo** [comment]
The 2 proposals of import are important when you try to use vuestic as dependency and not like base project ( cd https://github.com/epicmaxco/vuestic-admin/issues/30 ).
I've begin to do this, and that's work for the moment. I'll write a blog note or more developed documentation to do it simply.

**Comment by smartapant** [comment]
@atalargo wow, that's what really needed! We will try to find time and check this huge PR :)

**Comment by atalargo** [comment]
@smartapant thx :)

**Comment by atalargo** [comment]
I will push soon another commit in this PR for Dropdown (remove Bulma-expanding and replacing it by new  bootstrap 4 Dropdown)

**Comment by atalargo** [comment]
I have some other proposal not-relative imports and usage (icons, etc). Do you want i extract from this PR commit 732f4d3 and ac6e0cb to put them in another PR with other modification i should do for the same target?

**Comment by smartapant** [comment]
@atalargo yep, extracting them would be good

**Comment by smartapant** [comment]
@atalargo also could you please extract latest commits that you've made for Expand?

**Comment by atalargo** [comment]
Ok I'll do it soon

**Comment by Vad1mo** [comment]
what is the status of the PR @smartapant, @atalargo?

**Comment by smartapant** [comment]
@Vad1mo couldn't find time to check it :(

**Comment by atalargo** [comment]
@Vad1mo it's a "waiting" status :)

**Comment by zorn-v** [comment]
You should also change some classes in AuthLayout.
`hidden-lg-up` to `d-lg-none` for nav, and `hidden-md-down` to `d-none d-lg-flex` for wallpaper

**Comment by atalargo** [comment]
@zorn-v Fixed, i add 2 others fix for almost same css found in VuesticTabs

**Comment by Vad1mo** [comment]
So what is the status here on the PR, did you had time for a review?

File: package.json [hunk 1]
```diff
@@ -14,8 +14,8 @@
     "ammap3": "github:amcharts/ammap3",
     "awesome-bootstrap-checkbox": "1.0.0-alpha.5",
     "babel-polyfill": "^6.26.0",
-    "bootstrap": "4.0.0-alpha.6",
-    "bootstrap-vue": "^0.9.0",
+    "bootstrap": "^4.0.0-beta.2",
```
Comment (jean-moldovan) [inline package.json]: Lock the version please `"bootstrap": "4.0.0-beta.2"`

File: package.json [hunk 2]
```diff
@@ -14,8 +14,8 @@
...
+    "bootstrap-vue": "^1.1.0",
```
Comment (jean-moldovan) [inline package.json]: Looks like `bootstrap-vue` is not being used and could be safely removed.

File: src/components/vuestic-components/vuestic-tabs/VuesticTabs.vue [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <template>
   <div class="vuestic-tabs">
-    <div class="hidden-sm-down">
+    <div class="d-none d-sm-none d-md-flex">
```
Comment (jean-moldovan) [inline src/components/vuestic-components/vuestic-tabs/VuesticTabs.vue]: Hm, tabs are broken with this change..
@atalargo, could you please doublecheck?

**Comment by atalargo** [comment]
Sorry for the delay. I'll check all your remarks today or tomorrow.

**Comment by atalargo** [comment]
The commit titled "Update to last beta3" is just because bootstrap.beta3 is out, and there is not many changes, so i think could be a good idea to add to this pr. But i can remove it if you want

**Comment by atalargo** [comment]
I'll soon push a new commit with rebase on master and fix conflicting merge.

**Comment by atalargo** [comment]
I think all is ok now. (+ beta3)

**Comment by atalargo** [comment]
Oups. I missing "Can we make the buttons have same padding as before? (see http://vuestic.epicmax.co/#/ui/buttons)

Can we make alert success have the same background-color as before?"
I check this now. Sorry

**Comment by atalargo** [comment]
@jean-moldovan 
For your 1st point (ui buttons), I not see differences, except for the chevron down on one button in "Icons, Groups, Dropdowns" part. But this miss is from use of "ionicons": "^3.0.0", which didn't provide anymore this icon. ( cf. https://ionicframework.com/docs/ionicons/ ) I could do another pr for this fix if you want, choosing another icon (arrow-down is too small, i did the test, but maybe with another icons set like FontAwesome which provide a chevron.) 
So 2 questions: 
- Do you want i do another PR for the chevron or he is already fixed in another ticket? Do you want i add the fix in this PR (bad idea think)
- Which differences you see in ui/buttons appearance in this PR?

(I'll check your second point later)

**Comment by jean-moldovan** [comment]
Hey, glad you are back!

When I checked your branch (it had beta2 at the time) all buttons had no padding and looked really weird, could have been fixed in beta3.

Would be nice if you could add missing icon in a separate branch/PR.

Would be also nice if you could investigate why the 5th button turned green instead of being purple, but that's a small thing.

Thanks!

**Comment by atalargo** [comment]
Hi.
So it's the wanted behaviour/color, that 'success' state color are in purple? (i think i make a mistake with a sass variable)

**Comment by jean-moldovan** [comment]
No, should be green :)

**Comment by atalargo** [comment]
In the following screenshot, you can the actual in production (http://vuestic.epicmax.co/#/ui/buttons) current ui/buttons. As you can see, the "btn btn-success" button is ~purple (#db76df) 

So is it an error of the actual prod or should i try to put it in purple again (without transform alert-success that should be green yet -change in variables of sass of bootstrap) ?

capture d ecran 2018-01-10 a 14 41 32

**Comment by jean-moldovan** [comment]
Ah, thanks for pointing out!

It's a minor bug, no need to make it purple again 👍

**Comment by atalargo** [comment]
Ok cool :)

**Comment by atalargo** [comment]
I fix your second point (harder than i think) and a problem with background-color on striped table (and border of table row/col) too.
I think I do all you queries.

File: src/components/auth/signup/Signup.vue [hunk 4]
```diff
@@ -14,10 +14,10 @@
...
         </div>
       </div>
-      <div class="abc-checkbox abc-checkbox-primary">
-        <input id="checkbox1" type="checkbox" checked>
-        <label for="checkbox1">
-          <span class="abc-label-text">{{'auth.agree' | translate}} <router-link to="">{{'auth.termsOfUse' | translate}}</router-link></span>
+      <div class="form-check abc-checkbox abc-checkbox-primary">
+        <input class="form-check-input" id="checkbox1" type="checkbox" checked>
+        <label class="form-check-label" for="checkbox1">
+          <span class="abc-label-text">{{'auth.agree' | translate}} <router-link to="">Terms of Use.</router-link></span>
```
[the remark this answers is not in the document]
Comment (atalargo) [reply src/components/auth/signup/Signup.vue]: Oups. Miss it during rebase. Thanks

**Comment by smartapant** [comment]
@atalargo could you please sync pr with master? we've removed trouble-making package-lock.json.  This little thing - and ready to merge :)

**Comment by atalargo** [comment]
done :)

**Pull request merged**